dtfs-central-api/src/helpers/get-keying-sheet-calculation-tfm-facility-values.test.ts

```diff
@@ -0,0 +1,170 @@
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: today,
+          coverStartDate: null,
+        },
+      });
+    });
+
+    it('should throw a NotFoundError', async () => {
+      // Act / Assert
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the found tfm facility does not contain a cover end date', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue({
+        ...aTfmFacility(),
+        facilitySnapshot: {
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: null,
+          coverStartDate: today,
+        },
+      });
+    });
+
+    it('should throw a NotFoundError', async () => {
+      // Act / Assert
+      await expect(getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod)).rejects.toThrow(
+        new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the report period is provided', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacility);
+    });
+
+    it('should return a populated object', async () => {
+      const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod);
+
+      const expected = {
+        coverEndDate: convertTimestampToDate(today),
+        coverStartDate: convertTimestampToDate(today),
+        dayCountBasis: 365,
+        interestPercentage: 5,
+        coverPercentage: 5,
+        value: mockFacility.facilitySnapshot.value,
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the report period is not provided', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacility);
+    });
+
+    it("should return a populated object with the facility's cover end date", async () => {
+      const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId);
+
+      const expected = {
+        coverEndDate: convertTimestampToDate(today),
+        coverStartDate: convertTimestampToDate(today),
+        dayCountBasis: 365,
+        interestPercentage: 5,
+        coverPercentage: 5,
+        value: mockFacility.facilitySnapshot.value,
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the facility contains a completed amendment', () => {
+    const dateAfterReportPeriodEnd = new Date('2024-03-01');
```

Comment:
This date is not after the report period end at all, and if it were the test would fail. This is where I would suggest we mock the `getEffectiveCoverEndDateAmendment` function, but since that is not async I believe the decision is we shouldn't mock it so instead you need to maintain realistic data here please.

Reply:
changed to use addDays

Reply:
Resolving and adding a new comment because this is still incorrect